### PR TITLE
Added Action / Share extension to deeplink from mobile browsers

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1632

## What changed (plus any additional context for devs)
Added native iOS share/action extensions to handle deeplinking from popular web browsers (Safari, Chrome, DuckDuckGo, etc.)

**Testflight build 1.9.32 (21)**

## Screen recordings / screenshots
Screenshot 2024-07-18 at 12 29 54 PM

https://github.com/user-attachments/assets/a6849fde-49bd-4b63-befc-dbb482c8a344

## What to test

Ensure no regressions on Android

### CODE REVIEW

File: ios/Rainbow/AppDelegate.mm [hunk 1]
```diff
@@ -54,7 +54,7 @@ - (void)hideSplashScreenAnimated {
 
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
 {
- self.moduleName = @"Rainbow";
```
Comment (jinchung) [inline ios/Rainbow/AppDelegate.mm:57]: none of these changes are needed in this file yah ?

File: bitrise.yml
Comment (brunobar79) [inline bitrise.yml:1]: Pls revert the changes on this one 🙏

**Review by brunobar79 — APPROVED** [review]
Good stuff! 💯

**Comment by jinchung** [comment]
building a latest testflight to check after the latest fixes

**Review by jinchung — APPROVED** [review]
🌮
